File: proguard-services-mapper-maven/src/main/java/com/github/gianttreelp/proguardservicesmapper/maven/ProGuardServicesMapper.java
package com.github.gianttreelp.proguardservicesmapper.maven;

import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugins.annotations.LifecyclePhase;
import org.apache.maven.plugins.annotations.Mojo;
import org.apache.maven.plugins.annotations.Parameter;

import java.io.File;

import static com.github.gianttreelp.proguardservicesmapper.common.MappingKt.mapServices;

/**
 *  Maven plugin for ProGuard services mapping.
 *  <p>
 *      This plugin maps the services in the META-INF/services directory of a jar file to the corresponding
 *      obfuscated class names defined by the ProGuard mapping file.
 *  </p>
 */
@Mojo(
        name = ProGuardServicesMapper.ID,
        defaultPhase = LifecyclePhase.PACKAGE
)
public final class ProGuardServicesMapper extends AbstractMojo {
    static final String ID = "map-proguard";

    @Parameter(name = "input", required = true)
    private File input;

    @Parameter(name = "mapping", required = true)
    private File mapping;

    @Override
    public void execute() {
        mapServices(this.mapping.getAbsolutePath(), this.input.getAbsolutePath());
    }
}

File: proguard-services-mapper-maven/src/main/kotlin/ProGuardServicesMapper.kt
package de.gianttree.proguardservicesmapper.maven

import de.gianttree.proguardservicesmapper.common.mapServices
import org.apache.maven.plugin.AbstractMojo
import org.apache.maven.plugins.annotations.LifecyclePhase
import org.apache.maven.plugins.annotations.Mojo
import org.apache.maven.plugins.annotations.Parameter
import java.nio.file.Path

const val ID = "proguard-services-mapper"

@Mojo(
    name = ID,
    defaultPhase = LifecyclePhase.PACKAGE,
)
class ProGuardServicesMapper : AbstractMojo() {

    @Parameter(name = "$ID.input", required = true)
    private var inputPath: Path? = null

    @Parameter(name = "$ID.mapping", required = true)
    private var mappingPath: Path? = null

    override fun execute() {
        mapServices(inputPath.toString(), mappingPath.toString())
    }
}
